git_machete/cli.py

```diff
@@ -75,38 +78,172 @@ def usage(command: str = None) -> None:
         """[1:])))
 
 
-def short_usage() -> None:
-    print(
-        fmt("<b>Usage: git machete [--debug] [-h] [-v|--verbose] [--version] "
-            "<command> [command-specific options] [command-specific argument]</b>"))
+def short_usage() -> str:
+    return (fmt("<b>Usage: git machete [--debug] [-h] [-v|--verbose] [--version] "
+                "<command> [command-specific options] [command-specific argument]</b>"))
 
 
 def version() -> None:
     print(f"git-machete version {__version__}")
 
 
+def create_cli_parser() -> argparse.ArgumentParser:
+    cli_parser = argparse.ArgumentParser(
+        prog='git-machete', usage=short_usage(), argument_default=argparse.SUPPRESS)
+    cli_parser.add_argument('--debug', action='store_true')
+    cli_parser.add_argument(
+        '--version', action='version', version=f'%(prog)s version {__version__}')
+    cli_parser.add_argument('-v', '--verbose', action='store_true')
+
+    subparsers = cli_parser.add_subparsers(dest='command')
+
+    add_parser = subparsers.add_parser('add', argument_default=argparse.SUPPRESS)
+    add_parser.add_argument('branch', nargs='?')
+    add_parser.add_argument('-o', '--onto')
+    add_parser.add_argument('-R', '--as-root', action='store_true')
+    add_parser.add_argument('-y', '--yes', action='store_true')
+
+    advance_parser = subparsers.add_parser('advance')
+    advance_parser.add_argument('-y', '--yes', action='store_true', default=argparse.SUPPRESS)
+
+    anno_parser = subparsers.add_parser('anno')
+    # possible values of 'annotation_text' include: [], [''], ['some_val'], ['text_1', 'text_2']
+    anno_parser.add_argument('annotation_text', nargs='*')
+    anno_parser.add_argument('-b', '--branch', default=argparse.SUPPRESS)
+    anno_parser.add_argument(
+        '-H', '--sync-github-prs', action='store_true', default=argparse.SUPPRESS)
+
+    delete_unmanaged_parser = subparsers.add_parser('delete-unmanaged')
+    delete_unmanaged_parser.add_argument(
+        '-y', '--yes', action='store_true', default=argparse.SUPPRESS)
+
+    diff_full_parser = subparsers.add_parser('diff', aliases=['d'])
+    diff_full_parser.add_argument('branch', nargs='?')
+    diff_full_parser.add_argument(
+        '-s', '--stat', action='store_true', default=argparse.SUPPRESS)
+
+    discover_parser = subparsers.add_parser('discover', argument_default=argparse.SUPPRESS)
+    discover_parser.add_argument('-C', '--check-out-since')
```

Comment:
Hmmm how does argparse know that a parameter is required after `-C` and `-r` but not after `-l` or `-y`? or is it yet to be implemented?

Reply:
In `add_argument` method you can pass argument `action='store_true'`, which causes `parser` to put `my_optional_argument=True` into it's `argparse.Namespace` if added argument `is in` the parsed list. You can also do `'store_false'`.